src/main/java/net/minecraftforge/server/permission/nodes/PermissionDynamicContext.java

```diff
@@ -17,33 +17,32 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-package net.minecraftforge.server.permission.context;
+package net.minecraftforge.server.permission.nodes;
 
-import com.google.common.base.Preconditions;
-import net.minecraft.world.entity.player.Player;
-import net.minecraft.world.level.Level;
+import net.minecraft.util.StringRepresentable;
 
-import javax.annotation.Nullable;
-
-public class WorldContext extends Context
+/**
+ * Pair of a PermissionDynamic and a value of the corresponding type.
+ * Use {@link PermissionDynamic#createContext(StringRepresentable)} for constructing.
+ */
+public class PermissionDynamicContext<T extends StringRepresentable>
```

Comment:
I'd suggest to also take a look at this documentation after fixing the `PermissionDynamic` one, as there may be a need for some rewording.
I'd personally add that there is no restriction on how a permission handler should represent this context, so it can be in a blockstate format or whatever else.